base/audit_events.go

```diff
@@ -72,6 +240,638 @@ var AuditEvents = events{
+		},
+		EnabledByDefault:   true,
+		FilteringPermitted: false,
+		EventType:          eventTypeAdmin,
+	},
+	AuditIDClusterInfoRead: {
+		Name:               "Sync Gateway cluster info read",
+		Description:        "Sync Gateway cluster info was viewed",
+		EnabledByDefault:   true,
+		FilteringPermitted: false,
+		EventType:          eventTypeAdmin,
+	},
+	AuditIDCreateDatabase: {
+		Name:        "Create database",
+		Description: "A new database was created",
+		MandatoryFields: AuditFields{
+			"db":     "database name",
+			"config": "JSON representation of db config",
+		},
+		EnabledByDefault:   true,
+		FilteringPermitted: false,
+		EventType:          eventTypeAdmin,
+	},
+	AuditIDReadDatabase: {
+		Name:        "Read database",
+		Description: "Information about this database was viewed.",
+		MandatoryFields: AuditFields{
+			"db": "database name",
+		},
+		OptionalFields:     AuditFields{},
+		EnabledByDefault:   true,
+		FilteringPermitted: false,
+		EventType:          eventTypeUser,
+	},
+	AuditIDDeleteDatabase: {
+		Name:        "Delete database",
+		Description: "A database was deleted",
+		MandatoryFields: AuditFields{
+			"db": "database name",
+		},
+		EnabledByDefault:   true,
+		FilteringPermitted: false,
+		EventType:          eventTypeAdmin,
+	},
+	AuditIDReadDatabaseConfig: {
+		Name:        "Read database config",
+		Description: "Database configuration was viewed",
+		MandatoryFields: AuditFields{
+			"db": "database name",
+		},
+		EnabledByDefault:   true,
+		FilteringPermitted: false,
+		EventType:          eventTypeAdmin,
+	},
+	AuditIDUpdateDatabaseConfig: {
+		Name:        "Update database config",
+		Description: "Database configuration was updated",
+		MandatoryFields: AuditFields{
+			"db":     "database name",
+			"config": "JSON representation of new db config",
+		},
+		EnabledByDefault:   true,
+		FilteringPermitted: false,
+		EventType:          eventTypeAdmin,
+	},
+	AuditIDDatabaseOffline: {
+		Name:        "Database offline",
+		Description: "Database was taken offline",
+		MandatoryFields: AuditFields{
+			"db": "database name",
+		},
+		EnabledByDefault:   true,
+		FilteringPermitted: false,
+		EventType:          eventTypeAdmin,
+	},
+	AuditIDDatabaseOnline: {
+		Name:        "Database online",
+		Description: "Database was brought online",
+		MandatoryFields: AuditFields{
+			"db": "database name",
+		},
+		EnabledByDefault:   true,
+		FilteringPermitted: false,
+		EventType:          eventTypeAdmin,
+	},
+	AuditIDDatabaseCompactStatus: {
+		Name:        "Database compaction status",
+		Description: "Database compaction status was viewed",
+		MandatoryFields: AuditFields{
+			"db": "database name",
+		},
+		EnabledByDefault:   true,
+		FilteringPermitted: false,
+		EventType:          eventTypeAdmin,
+	},
+	AuditIDDatabaseCompactStart: {
+		Name:        "Database compaction start",
+		Description: "Database compaction was started",
+		MandatoryFields: AuditFields{
+			"db": "database name",
```

Comment:
This should say tombstone or compaction and dryrun parameters.

We can also reset a compaction, maybe this should be a separate event?

Reply:
We can reset only an attachment compaction, this means we'll start from fresh checkpoints.

Tombstone compaction is necessarily query based it always starts from scratch after it has been stopped.

We can do a dry run for attachment compaction? I wonder if we should have separate events for tombstone and attachment compaction and make more fields mandatory?

Reply:
I agree, separate events for attachment compaction and database compaction.